src/widget.ts

```diff
@@ -0,0 +1,52 @@
+import { Widget } from '@lumino/widgets';
+import { PromiseDelegate } from '@lumino/coreutils';
+import ImageEditor from 'tui-image-editor';
+
+
+export class ImageEditorWidget extends Widget {
+    private _editor: any;
+    private _ready = new PromiseDelegate<void>();
+
+    get editor(): any {
+        return this._editor;
+    }
+
+    get ready(): PromiseDelegate<void> {
+        return this._ready;
+    }
+
+    constructor(maxWidth: number, maxHeight: number, cornerSize: number, rotatingPointOffset: number) {
```

Comment:
You will receive the context
```suggestion
    constructor(context: DocumentRegistry.Context, maxWidth: number, maxHeight: number, cornerSize: number, rotatingPointOffset: number) {
```